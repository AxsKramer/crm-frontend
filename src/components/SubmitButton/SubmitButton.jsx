import React from 'react'

const SubmitButton = ({value, ...otherprops}) => <input 
  type='submit'
  className='btn btn-block bg-blue'
  value={value}
  {...otherprops}
/>

export default SubmitButton
