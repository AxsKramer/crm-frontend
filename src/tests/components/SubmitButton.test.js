import React from 'react';
import {shallow} from 'enzyme';
// import '@testing-library/jest-dom';

import SubmitButton from '../../components/SubmitButton/SubmitButton';

describe('Testing <SubmitButton />', () => {

  const value = 'Hello World';
  let wrapper = shallow(<SubmitButton value={ value } /> );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<SubmitButton value={ value } /> );
  })
  
  test('should be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should be only one component to render', () => {
    expect(wrapper.length).toEqual(1);
  })

  test('should be an input html element', () => {
    const isInputHTMLElement = wrapper.find('input').type('submit');
    expect(isInputHTMLElement).toBe('input');
  })

  test('should be an input with type submit', () => {
    const input = wrapper.find('input');
    expect(input.prop('type')).toBe('submit');
  })

  test('should have Hello World as value ', () => {
    const input = wrapper.find('input');
    expect(input.prop('value')).toBe(value);
  })

  test('should have btn class name', () => {
    const input = wrapper.find('input');
    const className = input.prop('className');
    expect(className.includes('btn')).toBe(true);
  })
})