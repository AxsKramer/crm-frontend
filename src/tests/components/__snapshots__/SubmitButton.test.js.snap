// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SubmitButton /> should be displayed correctly 1`] = `
<input
  className="btn btn-block bg-blue"
  type="submit"
  value="Hello World"
/>
`;
